desktop/ui/src/main/java/org/datacleaner/widgets/properties/SingleDatePropertySettingDialog.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.widgets.properties;
+
+import java.awt.BorderLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import java.util.Date;
+
+import javax.swing.ButtonGroup;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JRadioButton;
+import javax.swing.JTextField;
+
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.ImageManager;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.util.WidgetUtils;
+import org.datacleaner.util.convert.NowDate;
+import org.datacleaner.util.convert.ShiftedToday;
+import org.datacleaner.util.convert.TodayDate;
+import org.datacleaner.util.convert.YesterdayDate;
+import org.datacleaner.widgets.Alignment;
+import org.datacleaner.widgets.DescriptionLabel;
+import org.datacleaner.windows.AbstractDialog;
+import org.jdesktop.swingx.JXDatePicker;
+
+public class SingleDatePropertySettingDialog extends AbstractDialog implements ActionListener {
```

Comment:
The serializable class SingleDatePropertySettingDialog does not declare a static final serialVersionUID field of type long.